ports/mesa/portfile.cmake

```
@@ -0,0 +1,171 @@
+# Required LLVM modules: LLVM (modules: bitwriter, core, coroutines, engine, executionengine, instcombine, mcdisassembler, mcjit, scalaropts, transformutils) found: YES 
+
+#patches are from https://github.com/pal1000/mesa-dist-win/tree/master/patches
+set(PATCHES dual-osmesa.patch
+            dual-osmesa-part2.patch
+            swravx512.patch
+            )
+
+
+IF(VCPKG_TARGET_IS_WINDOWS)
+    set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled) # some parts of this port can only build as a shared library.
+endif()
+
+vcpkg_from_gitlab(
+    GITLAB_URL https://gitlab.freedesktop.org
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO mesa/mesa
+    REF  df2977f871fc70ebd6be48c180d117189b5861b5 #v20.2.2
+    SHA512 6c51d817fe265ea6405c4e8afbb516f30cf697d00cf39f162473ea8a59c202bcdfbfe4b6f7c4a6fd2d4e98eb4a1604cb5e0a02558338bf415e53fe5421cbfbbe
+    HEAD_REF master # branch name
+    PATCHES ${PATCHES} #patch name
+) 
+vcpkg_find_acquire_program(PYTHON3)
+get_filename_component(PYTHON3_DIR "${PYTHON3}" DIRECTORY)
+vcpkg_add_to_path("${PYTHON3_DIR}")
+vcpkg_add_to_path("${PYTHON3_DIR}/Scripts")
+if(DEFINED ENV{PYTHON})
+    set(ENV_PYTHON_BACKUP "$ENV{PYTHON}")
+endif()
+set(ENV{PYTHON} "${PYTHON3}")
+
+
+if (WIN32)
+    set(PYTHON_OPTION "")
+else()
+    set(PYTHON_OPTION "--user")
+endif()
+
+if(NOT EXISTS ${PYTHON3_DIR}/easy_install${VCPKG_HOST_EXECUTABLE_SUFFIX})
+    if(NOT EXISTS ${PYTHON3_DIR}/Scripts/pip${VCPKG_HOST_EXECUTABLE_SUFFIX})
+        vcpkg_from_github(
+            OUT_SOURCE_PATH PYFILE_PATH
+            REPO pypa/get-pip
+            REF 309a56c5fd94bd1134053a541cb4657a4e47e09d #2019-08-25
+            SHA512 bb4b0745998a3205cd0f0963c04fb45f4614ba3b6fcbe97efe8f8614192f244b7ae62705483a5305943d6c8fedeca53b2e9905aed918d2c6106f8a9680184c7a
+            HEAD_REF master
+        )
+        execute_process(COMMAND ${PYTHON3_DIR}/python${VCPKG_HOST_EXECUTABLE_SUFFIX} ${PYFILE_PATH}/get-pip.py ${PYTHON_OPTION})
+    endif()
+    execute_process(COMMAND ${PYTHON3_DIR}/Scripts/pip${VCPKG_HOST_EXECUTABLE_SUFFIX} install mako ${PYTHON_OPTION})
+    execute_process(COMMAND ${PYTHON3_DIR}/Scripts/pip${VCPKG_HOST_EXECUTABLE_SUFFIX} install setuptools ${PYTHON_OPTION})
+else()
+    execute_process(COMMAND ${PYTHON3_DIR}/easy_install${VCPKG_HOST_EXECUTABLE_SUFFIX} mako)
+    execute_process(COMMAND ${PYTHON3_DIR}/easy_install${VCPKG_HOST_EXECUTABLE_SUFFIX} setuptools)
+endif()
+if(NOT VCPKG_TARGET_IS_WINDOWS)
+    execute_process(COMMAND pip3 install setuptools mako)
+endif()
+
+vcpkg_find_acquire_program(FLEX)
+get_filename_component(FLEX_DIR "${FLEX}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${FLEX_DIR}")
+vcpkg_find_acquire_program(BISON)
+get_filename_component(BISON_DIR "${BISON}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${BISON_DIR}")
+
+if(WIN32) # WIN32 HOST probably has win_flex and win_bison!
+    if(NOT EXISTS "${FLEX_DIR}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")
+        file(CREATE_LINK "${FLEX}" "${FLEX_DIR}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")

```

This link should be created in a port-local location instead of mutating the tool's install. E.g:
```suggestion
        file(CREATE_LINK "${FLEX}" "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/flex${VCPKG_HOST_EXECUTABLE_SUFFIX}")
```

---

On Windows there is a very high probability that flex/bison are owned by vcpkg anyway.  I'll add a check if it is seated in `DOWNLOADS`